pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -524,21 +532,31 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 				set.Namespace,
 				set.Name,
 				replicas[target].Name)
-			err := ssc.podControl.DeleteStatefulPod(set, replicas[target])
+			if err := ssc.podControl.DeleteStatefulPod(set, replicas[target]); err != nil {
```

Comment:
shouldn't you add this pod to the list of unavailable pods? it's definitely absolutely about to become unavailable.

Reply:
that will catch up in the next iteration/requeue/resync, saves code, simplifis logic

Reply:
uh, but the current loop could easily kill far too many pods?

Controllers need to act on their prediction of the future state of the cluster.

Reply:
replied here https://github.com/kubernetes/kubernetes/pull/82162/files/45ec02a384b41abb86aaecd44fabb66e0e0f688c#r558754508

Reply:
@lavalamp can you take a look here, i answered this question, it ensures it will never kill too many pods

Reply:
See my response there: "That doesn't seem plausible since isHealthy is looking at a cached pod and not talking to apiserver again."

Reply:
@lavalamp this is fixed now, added this to unavailable with some comments explaining the reasoning

Reply:
this is resolved now